dom.go

```diff
@@ -125,8 +125,6 @@ type HTML struct {
 	key                             interface{}
 	// keyedChildren stores a map of keys to children, for keyed reconciliation.
 	keyedChildren map[interface{}]ComponentOrHTML
-	// hasKeyedChildren stores whether this element has keyed children.
-	hasKeyedChildren bool
```

Comment:
I couldn't find a valid reason to have this actually be a stateful field instead of just a variable / determined on the fly via `len(h.keyedChildren) > 0`. I find this much cleaner to read because if it's a var inside a func block you know it doesn't cross some call boundary.

It's possible I am just missing something though!

Reply:
The reason this is a field is because we delete keys from the prev render when we match them (so that we know which keys remain unmatched at the end of the render, to remove them), so we can't use `len(prev.keyedChildren)` to determine whether the previous render had keyed children or not, because they would all be deleted if we matched every previous key.

Reply:
Understood. I did not see that before. But I think we can still solve that without a separate `HTML` field, can we not? For example, why not just also declare `prevHasKeyedChildren` at the top of the func? I will push a commit for this now.

Reply:
Actually, this seems to be wrong, see my new comment at https://github.com/pdf/vecty/pull/2/files#r142022313